src/sas/qtgui/Calculators/GenericScatteringCalculator.py

```diff
@@ -1357,12 +1427,70 @@ def complete(self, input, update=None):
             out = numpy.hstack(out)
             self.data_to_plot = out
             logging.info('Gen computation completed.')
+
+        # if Beta(Q) Calculation has been requested, run calculation
+        if self.is_beta:
+            self.create_betaPlot()
+        
         self.cmdCompute.setText('Compute')
         self.cmdCompute.setToolTip("<html><head/><body><p>Compute the scattering pattern and display 1D or 2D plot depending on the settings.</p></body></html>")
         self.cmdCompute.clicked.disconnect()
         self.cmdCompute.clicked.connect(self.onCompute)
         self.cmdCompute.setEnabled(True)
         return
+    
+    def create_betaPlot(self):
+        """Carry out the compuation of beta Q using provided & calculated data
+        Returns a list of BetaQ values
+
+        """
+        
+        #Center Of Mass Calculation
+        CoM = self.centerOfMass()
+        self.data_betaQ = []
+
+        # Default values
+        xmax = self.qmax_x
+        xmin = self.qmax_x * _Q1D_MIN
+        qstep = self.npts_x
+
+        fQ = [0 for j in range(self.npts_x)]
+        currentQValue = []
+        for a in range(self.npts_x):
+            if(self.npts_x == 1):
+                currentQValue.append(xmin)
+            else: 
+                currentQValue.append(xmin + (xmax - xmin)/(self.npts_x-1)*a)
```

Comment:
This can be simplified by pulling the first case out of the loop. You want to use as few conditionals inside loops as possible (helps with vectorization).

Maybe something similar to:
```
currentQValue[0] = xmin
for a in range(self.npts_x-1):
    currentQValue.append(xmin + (xmax - xmin)/(self.npts_x-1)*(a+1))
```

Reply:
Thanks- this is a remnant of my previous program prior to being integrated into Sasview. Sasview only allows # of Q-Bins >= 2, so self.npts_x cannot be equal to 1. I have removed the test and reintegrated the code.